### PR TITLE
Add section to types and tags guide about schema validity

### LINKED ISSUE #726: Update Get Started and/or Types and Tags Guide to review schema invalidation
Having a brief overview of what a valid schema is and what types of operations could cause a schema to be invalidated would be helpful for new uses. Prior to the accessor release we should add some information about this, either to the Get Started Guide, the Types and Tags Guide or both.

### PR DESCRIPTION
**Pull request by tamargrey**
- Indicates how a Woodwork schema can be invalidated
- Closes #726

### CODE REVIEW

File: docs/source/guides/understanding_types_and_tags.ipynb [hunk 1]
```diff
@@ -476,7 +476,109 @@
...
+   "outputs": [],
+   "source": [
+    "df.ww.schema"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The Woodwork schema shown above can be seen reflected in the DataFrame below. Every column present in the schema is present in the DataFrame, the dtypes all match the physical types defined for each column, and the index column is both unique and represented on the DataFrame's underlying index."
```
Comment (thehomebrewnerd) [inline docs/source/guides/understanding_types_and_tags.ipynb:499]: Wondering if we should say "...the dtypes all match the physical types defined by each column's LogicalType..." here instead to make it a little more clear the physical type definition is not specified separate from the LogicalType?

Comment (thehomebrewnerd) [reply docs/source/guides/understanding_types_and_tags.ipynb:499]: Also for the index, maybe we could say "...and the Woodwork index column is both unique and matches the DataFrame's index."

File: docs/source/guides/understanding_types_and_tags.ipynb [hunk 2]
```diff
@@ -476,7 +476,109 @@
...
+   "outputs": [],
+   "source": [
+    "df.dtypes"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The question of what a valid schema is for a DataFrame is defined by Woodwork, and it is reliant on the DataFrame following Woodwork's type system. For this reason, it is not recommended to perform DataFrame operations directly on the DataFrame; instead, you should go through the `ww` namespace. If a DataFrame operation invalidates the Woodwork schema defined for that DataFrame, the typing information will be removed.\n",
```
Comment (thehomebrewnerd) [inline docs/source/guides/understanding_types_and_tags.ipynb:524]: How about "Woodwork defines the elements of a valid schema, and maintaining schema validity requires that the DataFrame follow Woodwork's type system." for the first sentence?

Also, maybe we could add a sentence prior to the last sentence to reinforce using the `ww` namespace: "Woodwork will attempt to retain a valid schema for any operations performed through the `ww` namespace."

File: docs/source/guides/understanding_types_and_tags.ipynb [hunk 3]
```diff
@@ -476,7 +476,109 @@
...
+    "\n",
+    "Therefore, when performing Woodwork operations, you can be sure that if the schema is present on `df.ww.schema` then the schema is valid for that DataFrame.\n",
+    "\n",
+    "### Defining a Valid Schema\n",
+    "\n",
+    "Given a DataFrame and its Woodwork typing information, the schema will be considered valid if:\n",
+    "- All of the columns present in the schema are present on the DataFrame and vice versa \n",
+    "- The physical type used by each column's Logical Type matches the corresponding series' `dtype`\n",
+    "- If an index is present, the index column is unique\n",
+    "- If an index is present, the DataFrame's underlying index matches the index column exactly\n",
```
Comment (thehomebrewnerd) [inline docs/source/guides/understanding_types_and_tags.ipynb:534]: Maybe we should clarify for pandas only?

**Review by thehomebrewnerd — COMMENTED** [review]
Looks pretty good. Just a few suggestions to consider.

File: docs/source/guides/understanding_types_and_tags.ipynb (hunk 1 above)
Comment (tamargrey) [reply docs/source/guides/understanding_types_and_tags.ipynb:499]: what do you think of saying `matches the DataFrame's underlying index` to follow the dichotomy of "Woodwork index" vs "DataFrame's underlying index", hopefully avoiding confusion

File: docs/source/guides/understanding_types_and_tags.ipynb (hunk 2 above)
Comment (tamargrey) [reply docs/source/guides/understanding_types_and_tags.ipynb:524]: That first sentence is much more clear, thanks!

File: docs/source/guides/understanding_types_and_tags.ipynb (hunk 1 above)
Comment (thehomebrewnerd) [reply docs/source/guides/understanding_types_and_tags.ipynb:499]: Yeah, that sounds good.

**Review by thehomebrewnerd — APPROVED** [review]
Looks good!

**Pull request merged**